packages/client/src/pages/register.vue

```diff
@@ -1,146 +1,233 @@
+
+          <q-btn
+            class="full-width"
+            color="primary"
+            :label="t('auth.register')"
+            type="submit"
+            :loading="isRegistering"
+            data-cy="submit-button"
+          />
         </q-form>
-      </q-card>
-    </div>
+      </q-card-section>
+
+      <template v-if="SOCIAL_LOGIN_ENABLED">
+        <q-separator />
+
+        <q-card-section class="column gap-8 no-padding">
+          <span class="text-black-87 text-center">
+            {{ t("common.or") }}
+          </span>
+
+          <social-auth-buttons type="register" />
+        </q-card-section>
+      </template>
+
+      <q-separator />
+
+      <q-card-section class="column gap-16 items-center no-padding">
+        <span class="text-dark text-subtitle1">
+          {{ t("auth.alreadyRegistered") }}
+        </span>
+
+        <q-btn
+          :label="t('auth.goToLogin')"
+          class="full-width outline-black-12"
+          outline
+          to="login"
+        />
+      </q-card-section>
+
+      <q-separator />
+
+      <router-link
+        :to="{ name: AvailableRouteNames.ForgotPassword }"
+        class="text-black-87 text-center text-subtitle1"
+        data-cy="forgot-password-link"
+      >
+        {{ t("auth.forgotPassword") }}
+      </router-link>
+    </q-card>
   </q-page>
 </template>
 
 <script setup lang="ts">
 import { ApolloError } from "@apollo/client/core";
-import { reactive, ref, toRaw } from "vue";
+import {
+  mdiEye,
+  mdiEyeOff,
+  mdiInformationOutline,
+} from "@quasar/extras/mdi-v7";
+import { QInputProps } from "quasar";
+import { computed, reactive, ref, toRaw } from "vue";
 import { useI18n } from "vue-i18n";
-import PasswordStrengthBar from "components/password-strength-bar.vue";
 import SocialAuthButtons from "components/social-auth-buttons.vue";
-import { RegisterPayload } from "src/@generated/graphql";
-import KPasswordInput from "src/components/k-password-input.vue";
-import { STRENGTH_BAR_STEPS } from "src/components/models";
+import { RegisterUserPayload } from "src/@generated/graphql";
+import { useTheme } from "src/composables/use-theme";
 import { notifyError } from "src/helpers/error-messages";
 import {
-  emailRule,
   makeValueMatchRule,
   requiredRule,
   validatePasswordRule,
 } from "src/helpers/rules";
+import { AvailableRouteNames } from "src/models/routes";
 import { useRegisterMutation } from "src/services/auth";
 
 const SOCIAL_LOGIN_ENABLED = process.env.SOCIAL_LOGIN_ENABLED === "true";
 
 const { t } = useI18n();
 
+const { theme } = useTheme();
+
 const showPassword = ref(false);
 
-const user = reactive<RegisterPayload>({
+type UserRegistrationData = Omit<
+  RegisterUserPayload,
+  "discount" | "retailLocationId" | "notes" | "dateOfBirth"
+> & {
+  confirmEmail: string;
+  dateOfBirth: string;
+};
+
+const user = reactive<UserRegistrationData>({
   email: "",
   firstname: "",
   lastname: "",
   password: "",
+  confirmEmail: "",
+  phoneNumber: "",
```

Comment:
Trying to register is telling me that the phone number does not exist inside the payload. Can you check this out? 
![image](https://github.com/ilmercatinodellibro/ilmercatinodellibro/assets/30588533/1b2bccec-1161-4ac3-9013-6d426bf983d6)